superset-frontend/src/dashboard/App.jsx

```diff
@@ -25,22 +25,28 @@ import { HTML5Backend } from 'react-dnd-html5-backend';
 import { DynamicPluginProvider } from 'src/components/DynamicPlugins';
 import setupApp from '../setup/setupApp';
 import setupPlugins from '../setup/setupPlugins';
-import DashboardContainer from './containers/Dashboard';
+import DashboardRoute from './components/DashboardPage';
 import { theme } from '../preamble';
 
 setupApp();
 setupPlugins();
 
-const App = ({ store }) => (
-  <Provider store={store}>
-    <DndProvider backend={HTML5Backend}>
-      <ThemeProvider theme={theme}>
-        <DynamicPluginProvider>
-          <DashboardContainer />
-        </DynamicPluginProvider>
-      </ThemeProvider>
-    </DndProvider>
-  </Provider>
-);
+const App = ({ store }) => {
+  const dashboardIdOrSlug = window.location.pathname.split('/')[3];
+  return (
+    <Provider store={store}>
+      <DndProvider backend={HTML5Backend}>
+        <ThemeProvider theme={theme}>
+          <DynamicPluginProvider>
```

Comment:
Maybe we need a `ProviderProvider` ;)

Reply:
Oh yeah! This should be addressed in the app merge PR.